### ISSUE TITLE
event() handler skips State.ACTIVE when used with <TapGestureHandler />

### ISSUE THREAD

**Issue by vpuonti** [software-mansion/react-native-reanimated#375]
Title.

Example code:

```ts

function interaction(tapState: Animated.Value<number>) {
  return cond(
    eq(tapState, State.ACTIVE),
    [debug('active', tapState)],
    [debug('inactive', tapState)]
  )
}

class Example extends React.Component {
  private tapState: Animated.Value<number> = new Value(State.UNDETERMINED)

  public render() {
    return (
      <TapGestureHandler
        onHandlerStateChange={event([
          {
            nativeEvent: {
              state: this.tapState,
            },
          },
        ])}
      >
        <Animated.View style={{ flex: 1 }}>
          <Animated.Code exec={interaction(this.tapState)} />
        </Animated.View>
      </TapGestureHandler>
    )
  }
}
```

Output to console.

```
inactive 0
inactive 2
inactive 5
inactive 2
inactive 5
inactive 2
inactive 5
inactive 2
inactive 5

```

**Comment by osdnk** [comment]
In my opinion, it might be considered as an expected behavior since it's a discrete gesture and a quick gesture state's changes in UIKit might not be sent to JS.

**Comment by alikazemkhanloo** [comment]
Hey, I can't handle ACTIVE state, how should I handle it?
is this issue resolved? I'm using version 1.7.0
if I use a function instead of `event()`, I can log ACTIVE state. but does this method cause performance issue calling the js thread to handle tap?

also I tried to use the state inside my code not even calling JS. but I couldn't see the result of my code in action. maybe sth is wrong with my code, but I couldn't find out anything wrong, I thought this issue might be relevant.
